Mathlib/MeasureTheory/Function/Jacobian.lean

```diff
@@ -766,6 +767,25 @@ theorem measurable_image_of_fderivWithin (hs : MeasurableSet s)
   haveI : DifferentiableOn ℝ f s := fun x hx => (hf' x hx).differentiableWithinAt
   hs.image_of_continuousOn_injOn (DifferentiableOn.continuousOn this) hf
 
+/-- If a function is differentiable and injective on a null measurable set,
+then the image is null measurable. -/
+theorem nullMeasurable_image_of_fderivWithin (hs : NullMeasurableSet s μ)
+    (hf' : ∀ x ∈ s, HasFDerivWithinAt f (f' x) s x) (hf : InjOn f s) :
+    NullMeasurableSet (f '' s) μ := by
+  rcases hs.exists_measurable_subset_ae_eq with ⟨t, ts, ht, t_eq_s⟩
+  have A : f '' s =ᵐ[μ] f '' t := by
```

Comment:
This argument occurs twice: do you think it's worth extracting into a helper lemma?

Reply:
I don't think so: it's really specific, so I don't expect it will ever find another use.